### DISCUSSION TITLE
How to handle websocket disconnect from client

### DISCUSSION THREAD

**Discussion by MichalPham** [encode/starlette#2543] [Q&A]
When disconectiong from client side (httpx-ws, js) using their respective `close` methods, the `startlette.websocket.receive_(text|bytes)` fails with `RuntimeError` and not with `WebSocketDisconnect`. Its caused by this [line](https://github.com/encode/starlette/blob/master/starlette/websockets.py#L51) because the message type is `websocket.close` and not `websocket.disconnect`. I have not found a way how to send `websocket.disconnect` looking at the implementation of starlette [close method](https://github.com/encode/starlette/blob/master/starlette/websockets.py#L202) it also sends `websocket.close`.

So the question really is how do i correctly disconect from client side.

**Comment by roeverni** [comment]
Hi,
I guess, I am facing the same issue. Are there any updates on this?

**Comment by dym-ok** [comment]
We seem to be affected by the same issue. There seems to be a little of a disconnect between message types used for connection termination, or am I confused?

**Comment by NiltonVolpato** [comment]
I'm also receiving:

RuntimeError: Expected ASGI message "websocket.receive" or "websocket.disconnect", but got 'websocket.close'

Is there a version we can rollback to?

**Reply by NiltonVolpato** [reply]
Actually this may be a problem with httpx-ws who is not following the ASGI spec.
